ospfd/ospf_ext.c

```diff
@@ -1125,7 +1125,8 @@ static int ospf_ext_pref_lsa_originate1(struct ospf_area *area,
 	if (IS_DEBUG_OSPF(lsa, LSA_GENERATE)) {
 		char area_id[INET_ADDRSTRLEN];
 
-		strncpy(area_id, inet_ntoa(area->area_id), INET_ADDRSTRLEN);
+		strncpy(area_id, inet_ntoa(area->area_id), sizeof(area_id) - 1);
+		area_id[sizeof(area_id) - 1] = 0;
```

Comment:
Looks good to me, but I'd prefer one of the following:
```
strlcpy(area_id, inet_ntoa(area->area_id), sizeof(area_id));
```
```
inet_ntop(AF_INET, &area->area_id, area_id, sizeof(area_id));
```
`strncpy()` is tricky and we should try to avoid it whenever we can.

Reply:
`inet_ntop()` is like `inet_ntoa()`, but instead of returning a pointer to a static buffer in libc, the user has to provide a buffer to be filled by the function (this way the function is reentrant).

To simplify the code, I think it's better to use `inet_ntop()` instead of `strlcpy()`+`inet_ntoa()` (not to mention it's one library call against two).

Reply:
> Ok, I understand the strlcpy vs strncpy. BTW, I don't get the "inet_ntop()" change (?)

You can avoid the `strXcpy()` entirely by just using the `inet_ntop()` function which will do the exact same thing.